### PR TITLE
[FIX JENKINS-38615] Add user to restart log message

### PR DESCRIPTION
**Pull request by daniel-beck**
`RestartJenkinsJob`, the checkbox one can tick where plugin download progress is shown, runs in a separate thread, so it doesn't have the current request's authentication. This fixes this to make Jenkins not log that anonymous restarted the instance.

@reviewbybees

### CODE REVIEW

**Review by batmat — APPROVED** [review]
:bee:

File: core/src/main/java/hudson/model/UpdateCenter.java [hunk 1]
```diff
@@ -1392,7 +1399,10 @@ public synchronized void run() {
             }
             status = new Running();
             try {
-                Jenkins.getInstance().safeRestart();
+                // safeRestart records the current authentication for the log, so set it to the managing user
+                try (ACLContext _ = ACL.as(User.get(authentication, false, Collections.emptyMap()))) {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/UpdateCenter.java]: I'm not sure it's going to work correctly if the User#get() method returns null.
May happen if a user gets deleted by somebody when hist session is active.

🐜

Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/UpdateCenter.java]: Agreed

**Comment by oleg-nenashev** [comment]
🐝 
@reviewbybees done

**Pull request merged**